contrib/aws/internal/eventbridge/eventbridge.go

```diff
@@ -0,0 +1,108 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package eventbridge
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge/types"
+	"github.com/aws/smithy-go/middleware"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"time"
+)
+
+const (
+	datadogKey      = "_datadog"
+	startTimeKey    = "x-datadog-start-time"
+	resourceNameKey = "x-datadog-resource-name"
+	maxSizeBytes    = 256 * 1024 // 256 KB
+)
+
+func EnrichOperation(span tracer.Span, in middleware.InitializeInput, operation string) {
+	switch operation {
+	case "PutEvents":
+		handlePutEvents(span, in)
+	}
+}
+
+func handlePutEvents(span tracer.Span, in middleware.InitializeInput) {
+	params, ok := in.Parameters.(*eventbridge.PutEventsInput)
+	if !ok {
+		log.Debug("Unable to read PutEvents params")
+		return
+	}
+
+	// Create trace context
+	carrier := tracer.TextMapCarrier{}
+	err := tracer.Inject(span.Context(), carrier)
+	if err != nil {
+		log.Debug("Unable to inject trace context: %s", err)
+		return
+	}
+
+	carrierJSON, err := json.Marshal(carrier)
+	if err != nil {
+		log.Debug("Unable to marshal trace context: %s", err)
+		return
+	}
+
+	// Prepare the reused trace context string
+	startTimeMillis := time.Now().UnixMilli()
+	reusedTraceContext := fmt.Sprintf(`%s,"%s":"%d"`, carrierJSON[:len(carrierJSON)-1], startTimeKey, startTimeMillis)
```

Comment:
Can you add a test that sets the trace prop style to `none`? I'm concerned in that case we'd end up with invalid json, like `{,"start":123}`.

I think the env var you want is `DD_TRACE_PROPAGATION_STYLE=none`.

Reply:
And actually, could probably avoid any such issues if you manually add the start time key to the carrier.

```go
carrier[startTimeKey] = strconv.Itoa(startTimeMillis)
carrierJSON, err := json.Marshal(carrier)
etc...
```

This way you're always guaranteed at least one item in the map.

Reply:
Good idea